networkx/algorithms/matching.py

```diff
@@ -41,18 +41,99 @@ def maximal_matching(G):
     The algorithm greedily selects a maximal matching M of the graph G
     (i.e. no superset of M exists). It runs in `O(|E|)` time.
     """
-    matching = set([])
-    nodes = set([])
-    for u,v in G.edges():
+    matching = set()
+    nodes = set()
+    for u, v in G.edges():
         # If the edge isn't covered, add it to the matching
         # then remove neighborhood of u and v from consideration.
-        if u not in nodes and v not in nodes and u!=v:
-            matching.add((u,v))
+        if u not in nodes and v not in nodes and u != v:
+            matching.add((u, v))
             nodes.add(u)
             nodes.add(v)
     return matching
 
 
+def matching_dict_to_set(matching):
+    """Converts a dictionary representing a matching (as returned by
+    :func:`max_weight_matching`) to a set representing a matching (as
+    returned by :func:`maximal_matching`).
+
+    """
+    # Need to compensate for the fact that both pairs (u, v) and (v, u)
+    # appear in `matching.items()`, so we sort each pair of nodes before
+    # adding it to the set.
+    return set(tuple(sorted(edge)) for edge in matching.items())
```

Comment:
It's even more subtle than that.  The 2-tuple being ordered "just" means we are using a Directed Edge to represent an Undirected Edge. The sortable feature is all about nodes. We want to allow the nodes to be arbitrary hashable... that is, not necessarily sortable. Maybe they are Graph objects?